documentation/tutorials/remember-me.md

```diff
@@ -0,0 +1,247 @@
+  tokens do
+    enabled? true
+    store_all_tokens? true
+    require_token_presence_for_authentication? true
+  end
+
+  addons do
+    ...
+    add_ons do
+      # Recommended: use the logout everywhere add on to revoke
+      # Remember Me tokens on password change.
+      log_out_everywhere do
+        apply_on_password_change? true
+      end
+    end
+  end
+
+  strategies do
+    ...
+    # Add the remember me strategy
+    remember_me do # Optionally name the strategy: `remember_me :remember_me do`
+      sign_in_action_name :sign_in_with_remember_me. # Optional defaults to :sign_in_with_[:strategy_name]
+      cookie_name :remember_me # Optional. Defaults to :remember_me
+      token_lifetime {30, :days} # Optional. Defaults to {30, :days}
+    end
+  end
+end
+```
+
+## Optionally add the Remember Me sign in action
+
+This action will be called by sign_in_with_remember_me plug to validate the remember me 
+token, and generate a token for the session.
+
+```elixir
+# lib/my_app/accounts/user.ex
+actions do
+  ...
+
+  # If not provided, a :sign_in_with_remember_me action will be 
+  # automagically created for you.
+  read :sign_in_with_remember_me do
+    description "Attempt to sign in using a remember me token."
+    get? true
+
+    argument :token, :string do
+      description "The remember me token."
+      allow_nil? false
+      sensitive? true
+    end
+
+    # validates the provided sign in token and generates a token
+    prepare AshAuthentication.Strategy.RememberMe.SignInPreparation
+
+    metadata :token, :string do
+      description "A JWT that can be used to authenticate the user."
+      allow_nil? false
+    end
+  end
+end
+```
+
+## Update your existing sign in actions to generate Remember Me tokens
+
+For each sign in action that might generate a Remember Me token, add an argument, 
+preparation, and metadata. We're going to add it to two actions: :sign_in_with_password 
+and :sign_in_with_token which is used by by AshAuthenticationPhoenix's liveview 
+password form.
+
+```elixir
+# lib/my_app/accounts/user.ex
+actions do
+  read :sign_in_with_password do
+    description "Attempt to sign in using a email and password."
+    get? true
+
+    argument :email, :ci_string do
+      description "The email to use for retrieving the user."
+      allow_nil? false
+    end
+
+    argument :password, :string do
+      description "The password to check for the matching user."
+      allow_nil? false
+      sensitive? true
+    end
+
+    # 1 of 3: Add the argument
+    argument :remember_me, :boolean do  # Optionally, use your own argument name
+      description "Whether to generate a remember me token."
+      allow_nil? true
+    end
+
+    prepare AshAuthentication.Strategy.Password.SignInPreparation
+
+    # 2 of 3: Add the preparation. See the preparation docs if using an argument
+    # or strategy name other than remember_me
+    prepare AshAuthentication.Strategy.RememberMe.MaybeGenerateTokenPreparation
```

Comment:
I assume this would need to be told what the new argument name is somehow if they don't use `remember_me`? Should we show that here?

Reply:
Updated code comments